corehq/apps/reports/standard/cases/basic.py

```diff
@@ -240,10 +241,20 @@ def rows(self):
 
     @property
     def json_response(self):
-        with self.profiler.timing_context if self.profiler_enabled else contextlib.nullcontext():
+        if not self.profiler_enabled:
+            return super().json_response
+
+        start_time = datetime.now()
+        with self.profiler.timing_context:
             response = super().json_response
 
-        if self.profiler_enabled:
-            # Todo: SC-4181
-            pass
+        elapsed_seconds = (datetime.now() - start_time).total_seconds()
+        if elapsed_seconds > 10:
+            self.profiler.timing_context.add_to_sentry_breadcrumbs()
+            request_dict = dict(self.request.GET.lists())
+
+            notify_exception(None, "LongRunningReport", details={
+                'request_dict': request_dict,
```

Comment:
are request parameters already available on sentry via the URL?

Additionally, do we need some backoff here so we don't report the same report multiple times?

Reply:
It seems like dimagi-utils takes as the first argument the `request` ([see here](https://github.com/dimagi/dimagi-utils/blob/8f5fe3fa59fd612770fad1ffecdbbf4e672e216b/dimagi/utils/logging.py#L30)), which in this case is `None`. 

I simply modelled the implementation after [this example](https://github.com/dimagi/commcare-hq/blob/601b40179e04c62d87907b869cb8c0b552fa2cc7/corehq/apps/ota/views.py#L162).

> Additionally, do we need some backoff here so we don't report the same report multiple times?

Do you mean if the user runs the same query for the same report multiple times?

Reply:
> It seems like dimagi-utils takes as the first argument the request ([see here](https://github.com/dimagi/dimagi-utils/blob/8f5fe3fa59fd612770fad1ffecdbbf4e672e216b/dimagi/utils/logging.py#L30)), which in this case is None.

Cool.

> Do you mean if the user runs the same query for the same report multiple times?

Yup. May be multiple users can also run the same query?